fn/option.go

```diff
@@ -226,10 +229,18 @@ func OptionToLeft[O, L, R any](o Option[O], r R) Either[O, R] {
 
 // OptionToRight can be used to convert an Option value into an Either, by
 // providing the Left value that should be used if the Option value is None.
-func OptionToRight[O, L, R any](o Option[O], l L) Either[L, O] {
+func OptionToRight[O, L any](o Option[O], l L) Either[L, O] {
 	if o.IsSome() {
 		return NewRight[L, O](o.some)
 	}
 
 	return NewLeft[L, O](l)
 }
+
+// Note allows you to convert an Option value to a result with your own error
+// message.
+func (o Option[A]) Note(errString string, args ...interface{}) Result[A] {
```

Comment:
Just pass in `e error`?

Reply:
The main use case here is when something that ordinarily returns None needs a new error attached to it. We *can* pass an error here but I think it's less useful than this. Perhaps I can do both `Note` and `Notef` variants.

Reply:
Ah gotcha, don't have very strong feelings here so happy to leave as-is 👍

Reply:
I split it into Note and Notef